### PR TITLE
Added ability to configure Ajv instance via callback

### PR DESCRIPTION
**Pull request by lvl99**
I had need of adding some custom formats and keywords to the Ajv instance to test some of my validation schemas. While I found I could add custom formats easily through the `ajvOptions.formats` property, adding custom keywords through `ajvOptions` was not possible.

I put in the ability to trigger a callback with the new Ajv instance so that developers could further extend and customise Ajv for their testing purposes.

Additionally, I also put the instantiation before the returned `toMatchSchema` and `toBeValidSchema` functions as creating a new instance of Ajv on each function call might be more work than necessary? Happy to roll that change back if it's not necessary.

### CODE REVIEW

**Comment by lvl99** [comment]
Seems the CI is not passing due to `npm install` not being run. Is that normal?

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6699 @@
+{
```
Comment (10xLaCroixDrinker) [inline package-lock.json]: Remove this file

File: __tests__/matchers/toMatchSchema.spec.js [hunk 2]
```diff
@@ -134,10 +144,38 @@ describe('toMatchSchema', () => {
...
         });
       });
     });
   });
+
+  describe('custom keywords', () => {
+    it('typeof', () => {
+      expect({}).toMatchSchemaUnderTest({
+        typeof: 'object',
+      });
```
Comment (10xLaCroixDrinker) [inline __tests__/matchers/toMatchSchema.spec.js]: add some tests to see that failures throw errors

File: index.js [hunk 3]
```diff
@@ -16,25 +16,26 @@ const chalk = require('chalk');
...
-    toMatchSchema: buildToMatchSchema(options),
-    toBeValidSchema: buildToBeValidSchema(options),
+    toMatchSchema: buildToMatchSchema(options, afterAjvInitialized),
+    toBeValidSchema: buildToBeValidSchema(options, afterAjvInitialized),
   };
 }
 
 module.exports.matchers = matchersWithOptions();
-module.exports.matchersWithFormats = (formats = {}) => {
+module.exports.matchersWithFormats = (formats = {}, afterAjvInitialized) => {
```
Comment (10xLaCroixDrinker) [inline index.js]: since `matchersWithFormats` is deprecated, dont add to its API

File: index.js [hunk 4]
```diff
@@ -16,25 +16,26 @@ const chalk = require('chalk');
 const buildToMatchSchema = require('./matchers/toMatchSchema');
 const buildToBeValidSchema = require('./matchers/toBeValidSchema');
 
-function matchersWithOptions(userOptions = {}) {
+function matchersWithOptions(userOptions = {}, afterAjvInitialized) {
```
Comment (10xLaCroixDrinker) [inline index.js]: I think `extendAjv` would be a better variable name here (same applies to usage in matchers)

File: __tests__/matchers/toMatchSchema.spec.js [hunk 5]
```diff
@@ -12,17 +12,27 @@
  * the License.
  */
 
+const ajvKeywords = require('ajv-keywords');
 const chalk = require('chalk');
 const toMatchSchemaUnderTest = require('../..').matchers.toMatchSchema;
-const toMatchSchemaWithFormatsUnderTest = require('../..').matchersWithFormats({
-  bcp47: /^[a-z]{2}-[A-Z]{2}$/,
+const toMatchSchemaWithOptionsUnderTest = require('../..').matchersWithOptions({
```
Comment (10xLaCroixDrinker) [inline __tests__/matchers/toMatchSchema.spec.js]: `matchersWithOptions` should be tested separately from `matchersWithFormats`

File: index.js (hunk 4 above)
Comment (lvl99) [reply index.js]: Yes, much better naming suggestion!

**Comment by lvl99** [comment]
@10xLaCroixDrinker it should be OK to re-review now :+1:

**Comment by lvl99** [comment]
@10xLaCroixDrinker are you OK to review this any time soon? I've got some other code that's hoping it will be accepted and released 🤞 🙏

**Pull request merged**